mapie/futur/split/regression.py

```diff
@@ -0,0 +1,231 @@
+
+        WARNING: If ``random_state``is not ``None``, ``np.random.seed`` will
+        be changed, which will reset the seed for all the other random
+        number generators. It may have an impact on the rest of your code.
+
+        By default ``None``.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.regression import SplitCPRegressor
+    >>> np.random.seed(1)
+    >>> X_train = np.arange(0,400, 2).reshape(-1, 1)
+    >>> y_train = 2*X_train[:,0] + np.random.rand(len(X_train))
+    >>> mapie_reg = SplitCPRegressor(alpha=0.1, random_state=1)
+    >>> mapie_reg = mapie_reg.fit(X_train, y_train)
+    >>> y_pred, y_pis = mapie_reg.predict(X_train)
+    """
+    def __init__(
+        self,
+        predictor: Optional[
+            Union[
+                RegressorMixin,
+                Pipeline,
+                List[Union[RegressorMixin, Pipeline]]
+            ]
+        ] = None,
+        calibrator: Optional[BaseCalibrator] = None,
+        cv: Optional[
+            Union[str, BaseCrossValidator, BaseShuffleSplit]
+        ] = None,
+        alpha: Optional[float] = None,
+        conformity_score: Optional[ConformityScore] = None,
+        random_state: Optional[int] = None,
+    ) -> None:
+        self.random_state = random_state
+        self.cv = cv
+        self.predictor = predictor
+        self.conformity_score = conformity_score
+        self.calibrator = calibrator
+        self.alpha = alpha
+
+    def _check_fit_parameters(self) -> RegressorMixin:
+        """
+        Check and replace default value of ``predictor`` and ``cv`` arguments.
+        Copy the ``predictor`` in ``predictor_`` attribute if ``cv="prefit"``.
+        """
+        self.cv = self._check_cv(self.cv)
+        predictor = check_estimator_regression(self.predictor, self.cv)
+        return predictor
+
+    def _check_calibrate_parameters(self) -> Tuple[
+        ConformityScore, BaseCalibrator
+    ]:
+        """
+        Check and replace default ``conformity_score``, ``alpha`` and
+        ``calibrator`` arguments.
+        """
+        conformity_score_ = check_conformity_score(
+            self.conformity_score, self.default_sym_
+        )
+        calibrator = check_calibrator(self.calibrator)
+        self._check_alpha(self.alpha)
+        calibrator.sym = conformity_score_.sym
+        calibrator.alpha = self.alpha
+        calibrator.random_state = self.random_state
+        return conformity_score_, calibrator
+
+    def predict_score(
+        self, X: ArrayLike
+    ) -> NDArray:
+        """
+        Compute the predictor prediction, used to compute the
+        conformity scores.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, )
+            predictions
+        """
+        return self.predictor_.predict(X)
+
+    def predict_bounds(
```

Comment:
Why is this not simply called `predict`?

Reply:
Because ``SlpitCP`` already have a ``predict`` method, which call : 
```python
y_pred = self.predict_score(X)
check_is_fitted(self, self.calib_attributes)
bounds_arguments = self.get_method_arguments(
     self.calibrator_.predict, {}, kwargs,
)
y_bounds = self.predict_bounds(X, y_pred, **bounds_arguments)
return self.predict_best(y_pred), y_bounds
```
Once again, it allow the mother class ``SplitCP`` to be compatible with classification (which require ``y_pred_proba``)